dt-bindings: u-boot: Add variables for bootscript location

michalsimek opened the pull request:
Add bootscr-address and offset-from-ram-start properties to help in easier picking of boot script file when automated flows are used.

The bootscr-address holds the full 64 bit address of the bootscript file.

The bootscr-offset-from-ram-start holds the offset address of the bootscript file from the start of the ram base in systems with RAM detection.

sjg20 commented on dtschema/schemas/options/u-boot.yaml:
```diff
@@ -71,6 +71,21 @@ properties:
+    default: 0x0
+    description:
+      This property holds the full address of the boot script file. This helps in
+      making automated flows easier by fetching address directly from DT.
```
Can you add a bit more detail, or point to docs in the U-Boot tree? How does the script end up in memory at that address? In what circumstances is it actually run?

michalsimek replied:
In our implementation scriptaddr variable is setup based on DT when variables are not saved by SOC. 
And scriptaddr is mandatory variable for distro boot and store address where boot.scr script is loaded prior execution.
https://github.com/u-boot/u-boot/blob/master/doc/develop/distro.rst?plain=1#L291

It shouldn't be a problem to extend documentation and point to exact variable name and it's usage.

sjg20 replied:
OK I see. We are moving away from distroboot and to standard boot.

I agree it makes sense to put these variables in the DT rather than the environment, at least as a default.

In that case how about having a subnode in options, like:

options {
   bootstd {
      compatible = "u-boot,bootstd";
      script-addr = /bits 64/ <...>;
      script-ram-offset =...
   }

as I suspect we will end up adding all the other env vars in here too.

michalsimek replied:
Why this should be different compatible if we have u-boot,config already? 
For example bootdelay-sec or bootcmd are also related to boot process. Does it mean that we should move it here too? 

Pretty much all of these properties should be moved here. 
https://github.com/u-boot/u-boot/blob/master/doc/device-tree-bindings/config.txt

michalsimek commented on dtschema/schemas/options/u-boot.yaml:
```diff
@@ -71,6 +71,21 @@ properties:
       2: use simplified command line (e.g. avoid hush)
       3... reserved
 
+  bootscr-address:
```
Implementation we did that properties are alternative.

https://lore.kernel.org/r/6d68ea2df73b206af10a89b6a61a09d688fb6c94.1686665564.git.michal.simek@amd.com

michalsimek commented:
Failures are in installing dependencies phase. 
Based on discussion with @sjg20 I have renamed offset property to bootscr-ram-offset.
Fixed property type and ensure that only one property is valid at a time with updating example.

sjg20 commented on dtschema/schemas/options/u-boot.yaml:
```diff
@@ -71,6 +71,21 @@ properties:
+    $ref: /schemas/types.yaml#/definitions/uint64
+    default: 0x0
+    description:
+      This property holds the full address of the boot script file. This helps in
```
nit: s/This property holds/Holds/

Does this get put into the scriptaddr variable? If not, what is the relationship between these two? I think this needs more detail about how to use it. You have written it in the comments below, so please add it here

sjg20 commented on dtschema/schemas/options/u-boot.yaml:
```diff
@@ -71,6 +71,21 @@ properties:
+    $ref: /schemas/types.yaml#/definitions/uint64
+    default: 0x0
+    description:
+      This property holds the boot script file offset from start of the ram base
```
Again you shouldn't say 'This property' since we know it is a property :-)

sjg20 replied:
also again please mention where this address ends up and what it affects. The bindings just need more detail so that people can understand it a bit better (what it is for, how it works, why I might use it, etc.)

sjg20 commented on dtschema/schemas/options/u-boot.yaml:
```diff
@@ -71,6 +71,28 @@ properties:
+    default: 0x0
+    description:
+      Holds the boot script file offset from the start of the ram base address.
+      Platforms with run time RAM detection algorithms should use this property
```
run-time RAM-detection

michalsimek replied:
fixed.

sjg20 reviewed (approved):
LGTM but please fix nit

michalsimek commented:
@robherring: Can you please merge it?

robherring commented:
The property names in the commit msg don't match the actual name.

michalsimek commented:
@robherring commit message updated.
